### ISSUE TITLE
Inductor vectorized GELU kernel produces tiny negative value (~-1.65e-7) where eager produces -0.0, causing sign() mismatch

### ISSUE THREAD

**Issue by XYZboom** [pytorch/pytorch#189804]
### 🐛 Describe the bug

`torch.compile` with the `inductor` backend produces a numerical discrepancy in `F.gelu(x)` for input values near `x ≈ -5.53`. In eager mode, `F.gelu(-5.53)` produces `-0.0` (negative zero, 0x80000000). In inductor, it produces a tiny negative value (`≈ -1.65e-7`). When followed by `torch.sign()`, this causes `sign(-0.0) = 0` in eager vs `sign(tiny_negative) = -1` in inductor — a difference of 1.0 in the output.
In the original fuzzer-generated model, the chain `log2(x) → log2(result) → div → gelu → sign → sigmoid` produces `max_diff=0.23` at output index 13 (shape [23, 64, 62]).

```python
import torch
import torch.nn.functional as F

# Values from the original computation chain where gelu produces -0.0
x = torch.tensor([-5.533906936645508, -5.531643390655518], dtype=torch.float32)

class BugModel(torch.nn.Module):
    def forward(self, x):
        return torch.sign(F.gelu(x))

model = BugModel()

with torch.no_grad():
    ref = model(x)  # eager

with torch.no_grad():
    aot_out = torch.compile(model, backend="aot_eager")(x)

with torch.no_grad():
    ind_out = torch.compile(model, mode="default")(x)

print(f"Input:  {x.tolist()}")
print(f"Eager:  {ref.tolist()}   (gelu produces -0.0, sign(-0.0)=0)")
print(f"aot:    {aot_out.tolist()}")
print(f"Induct: {ind_out.tolist()}   (gelu produces ~-1.65e-7, sign(tiny_neg)=-1)")
print(f"aot vs ref: same={torch._dynamo.utils.same(ref.double(), aot_out.double())}")
print(f"ind  vs ref: same={torch._dynamo.utils.same(ref.double(), ind_out.double())}")
print(f"max_diff (ind vs ref): {(ref.double() - ind_out.double()).abs().max().item():.6f}")

# Also the full chain from the original source
print("\n=== Full chain from original source ===")
torch.manual_seed(42)
v6 = torch.randn(23, 64, 62, dtype=torch.float32, device="cpu")

class FullChainModel(torch.nn.Module):
    def forward(self, x):
        a = torch.log2(x)
        b = torch.log2(a)
        c = torch.div(a, b)
        d = F.gelu(c)
        e = torch.sign(d)
        return torch.sigmoid(e)

full_model = FullChainModel()
with torch.no_grad():
    full_ref = full_model(v6)
full_ind = torch.compile(full_model, mode="default")
with torch.no_grad():
    full_ind_out = full_ind(v6)

print(f"Full chain ind vs ref: same={torch._dynamo.utils.same(full_ref.double(), full_ind_out.double())}")
print(f"max_diff: {(full_ref.double() - full_ind_out.double()).abs().max().item():.6f}")
```
topic: fuzzer

## Expected behavior

```
Eager:  [0.0, 0.0]   (sign(-0.0) = 0)
aot:    [0.0, 0.0]   (same as eager)
Induct: [0.0, 0.0]   (should also be 0.0)
```

## Actual behavior

```
E0714 09:07:13.167000 640949 site-packages/torch/_dynamo/utils.py:3706] RMSE (res-fp64): 1.00000, (ref-fp64): 0.00000 and shape=torch.Size([2]). res.dtype: torch.float64, multiplier: 3.000000, tol: 0.000100, use_larger_multiplier_for_smaller_tensor: 0
Input:  [-5.533906936645508, -5.531643390655518]
Eager:  [0.0, 0.0]   (gelu produces -0.0, sign(-0.0)=0)
aot:    [0.0, 0.0]
Induct: [-1.0, -1.0]   (gelu produces ~-1.65e-7, sign(tiny_neg)=-1)
aot vs ref: same=True
ind  vs ref: same=False
max_diff (ind vs ref): 1.000000

=== Full chain from original source ===
E0714 09:07:13.188000 640949 site-packages/torch/_dynamo/utils.py:3706] RMSE (res-fp64): 0.00153, (ref-fp64): 0.00000 and shape=torch.Size([23, 64, 62]). res.dtype: torch.float64, multiplier: 2.000000, tol: 0.000100, use_larger_multiplier_for_smaller_tensor: 0
Full chain ind vs ref: same=False
max_diff: 0.231059

```

### Versions

Collecting environment information...
PyTorch version: 2.13.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.4 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: 15.0.7
CMake version: version 3.28.3
Libc version: glibc-2.39

Python version: 3.11.15 (main, Jun 11 2026, 15:20:16) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-6.17.0-35-generic-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False
Caching allocator config: N/A

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           48 bits physical, 48 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  32
On-line CPU(s) list:                     0-31
Vendor ID:                               AuthenticAMD
Model name:                              AMD Ryzen 9 9950X 16-Core Processor
CPU family:                              26
Model:                                   68
Thread(s) per core:                      2
Core(s) per socket:                      16
Socket(s):                               1
Stepping:                                0
Frequency boost:                         enabled
CPU(s) scaling MHz:                      79%
CPU max MHz:                             5756.4521
CPU min MHz:                             624.1940
BogoMIPS:                                8599.99
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good amd_lbr_v2 nopl xtopology nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpuid_fault cpb cat_l3 cdp_l3 hw_pstate ssbd mba perfmon_v2 ibrs ibpb stibp ibrs_enhanced vmmcall fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local user_shstk avx_vnni avx512_bf16 clzero irperf xsaveerptr rdpru wbnoinvd cppc arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif x2avic v_spec_ctrl vnmi avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq rdpid bus_lock_detect movdiri movdir64b overflow_recov succor smca fsrm avx512_vp2intersect flush_l1d amd_lbr_pmc_freeze
Virtualization:                          AMD-V
L1d cache:                               768 KiB (16 instances)
L1i cache:                               512 KiB (16 instances)
L2 cache:                                16 MiB (16 instances)
L3 cache:                                64 MiB (2 instances)
NUMA node(s):                            1
NUMA node0 CPU(s):                       0-31
Vulnerability Gather data sampling:      Not affected
Vulnerability Ghostwrite:                Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Old microcode:             Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Mitigation; IBPB on VMEXIT only
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; STIBP always-on; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Not affected
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Mitigation; IBPB on VMEXIT

Versions of relevant libraries:
[pip3] numpy==2.4.6
[pip3] nvidia-cublas==13.1.1.3
[pip3] nvidia-cuda-cupti==13.0.85
[pip3] nvidia-cuda-nvrtc==13.0.88
[pip3] nvidia-cuda-runtime==13.0.96
[pip3] nvidia-cudnn-cu13==9.20.0.48
[pip3] nvidia-cufft==12.0.0.61
[pip3] nvidia-curand==10.4.0.35
[pip3] nvidia-cusolver==12.0.4.66
[pip3] nvidia-cusparse==12.6.3.3
[pip3] nvidia-cusparselt-cu13==0.8.1
[pip3] nvidia-nccl-cu13==2.29.7
[pip3] nvidia-nvjitlink==13.3.33
[pip3] nvidia-nvtx==13.0.85
[pip3] torch==2.13.0
[pip3] triton==3.7.1
[conda] numpy                     2.4.6                    pypi_0    pypi
[conda] nvidia-cublas             13.1.1.3                 pypi_0    pypi
[conda] nvidia-cuda-cupti         13.0.85                  pypi_0    pypi
[conda] nvidia-cuda-nvrtc         13.0.88                  pypi_0    pypi
[conda] nvidia-cuda-runtime       13.0.96                  pypi_0    pypi
[conda] nvidia-cudnn-cu13         9.20.0.48                pypi_0    pypi
[conda] nvidia-cufft              12.0.0.61                pypi_0    pypi
[conda] nvidia-curand             10.4.0.35                pypi_0    pypi
[conda] nvidia-cusolver           12.0.4.66                pypi_0    pypi
[conda] nvidia-cusparse           12.6.3.3                 pypi_0    pypi
[conda] nvidia-cusparselt-cu13    0.8.1                    pypi_0    pypi
[conda] nvidia-nccl-cu13          2.29.7                   pypi_0    pypi
[conda] nvidia-nvjitlink          13.3.33                  pypi_0    pypi
[conda] nvidia-nvtx               13.0.85                  pypi_0    pypi
[conda] torch                     2.13.0                   pypi_0    pypi
[conda] triton                    3.7.1                    pypi_0    pypi

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @aditew01 @chauhang @penguinwu @voznesenskym @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by frost-intel** [comment]
The root cause here is the vectorized  `erf()` which doesn't numerically match `std::erf`. This particular case is very ill-conditioned input (GELU followed by sign) but it doesn't really represent an accuracy problem. Any attempt to close the gap would likely involve a major slowdown for vectorized code which is a non-starter.  @XYZboom Do you feel differently?

@malfet I think we can close this as not-planned?

**Comment by XYZboom** [comment]
@frost-intel Thank you for your reply. I agree with that perspective.
